pkg/stacks/unpack.go

```diff
@@ -519,8 +541,9 @@ func NewStackPackage(baseDir, tmplCtrlImage string) *StackPackage {
 //
 // baseDir is expected to be an absolute path, i.e. have a root to the path,
 // at the very least "/".
-func Unpack(rw walker.ResourceWalker, out io.StringWriter, baseDir, permissionScope string, tsControllerImage string) error {
-	sp := NewStackPackage(filepath.Clean(baseDir), tsControllerImage)
+func Unpack(rw walker.ResourceWalker, out io.StringWriter, baseDir, permissionScope string, tsControllerImage string, log logging.Logger) error {
+	l := log.WithValues("operation", "unpack")
```

Comment:
I was thinking about this earlier today! Glad to see unpack getting logs.

Is there any configuration that would result in these log messages disrupting the yaml in the pod log that we unpacked?

Reply:
in general, logging from the unpacking job will make the creation of the unpacked objects fail.  The expectation by the stack manager is that it can take the entirety of the unpack job's logs (stdout) and send them directly to the k8s api server.  If log statements are intermixed with that output of object yaml's, then something in the create flow will fail.

This seems to already been known and "documented" in https://github.com/crossplane/crossplane/blob/master/pkg/controller/stacks/install/installjob.go#L109-L111, so perhaps the policy that all logging in the unpack job should be at Debug level will work for us.

Reply:
Is there a way to configure the logger to prepend all message newlines with `#` so they are ignored by the yaml parser?

Reply:
https://github.com/kubernetes/kubernetes/issues/28167 seems related (asks for the log output to be stderr vs stdout identifiable) but I don't see any answers there.

Reply:
https://blog.sandipb.net/2018/05/03/using-zap-creating-custom-encoders/ has an example of using a custom log encoder.

Reply:
Yeah - it'd be great to have a way to log and have the job be functional, but having logs we can enable is a good first step. I do not plan to do a longer-term solution in this PR